src/main/java/fi/aalto/cs/apluscourses/presentation/base/FilterEngine.java

```diff
@@ -0,0 +1,83 @@
+package fi.aalto.cs.apluscourses.presentation.base;
+
+import fi.aalto.cs.apluscourses.presentation.filter.Filter;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+public class FilterEngine {
```

Comment:
Hmm... Now when we look at this class, it has very little to do with _filtering_ anymore. More suitable name for it could be e.g. "LastComeServedExecutor" or something like that. What do you think? Would you mind changing that?